src/common/transformations/src/transformations/common_optimizations/variadic_split_merge.cpp

```diff
@@ -0,0 +1,347 @@
+            } else if (ov::is_type<ov::op::v8::Slice>(node)) {
+                auto slice = ov::as_type_ptr<ov::op::v8::Slice>(node);
+
+                if (slice->get_input_size() >= 5) {
+                    auto axis_input = slice->get_input_node_shared_ptr(4);
+                    auto axis_ret = get_scalar(axis_input);
+                    if (axis_ret.err)
+                        return false;
+
+                    splitlist[i].axis = axis_ret.value;
+                }
+
+                auto step_input = slice->get_input_node_shared_ptr(3);
+                auto step_ret = get_scalar(step_input);
+                if (step_ret.err)
+                    return false;
+
+                if (step_ret.value != 1)
+                    return false;
+            }
+        }
+
+        auto axis = splitlist[0].axis;
+
+        // All axis have to be the same
+        for (size_t i = 1; i < splitlist.size(); i++) {
+            auto split = splitlist[i];
+            if (split.axis != axis)
+                return false;
+        }
+
+        int64_t total_len = 0;
+
+        // Axis partial shape has to be static
+        for (auto split : splitlist) {
+            auto node = split.node;
+
+            auto input_ps = node->get_input_partial_shape(0);
+            auto output_ps = node->get_output_partial_shape(0);
+
+            if (!input_ps[axis].is_static() || !output_ps[axis].is_static())
+                return false;
+
+            auto& total_length = input_ps[axis];
+            total_len = total_length.get_length();
+        }
+
+        // Get begin and end
+        for (auto& split : splitlist) {
+            auto node = split.node;
+            if (ov::is_type<ov::op::v1::StridedSlice>(node)) {
+                auto slice = ov::as_type_ptr<ov::op::v1::StridedSlice>(node);
+
+                auto begin_input = slice->get_input_node_shared_ptr(1);
+                auto end_input = slice->get_input_node_shared_ptr(2);
+                if (slice->get_input_size() > 3) {
+                    auto stride_input = slice->get_input_node_shared_ptr(3);
+                    auto stride_const = ov::as_type_ptr<ov::op::v0::Constant>(stride_input);
+                    if (!stride_const)
+                        return false;
+
+                    auto stride_vec = stride_const->cast_vector<int64_t>();
+                    auto input_ps = node->get_input_partial_shape(0);
+                    uint64_t rank = input_ps.rank().get_length();
+
+                    if (stride_vec.size() != rank)
+                        return false;
+
+                    if (stride_vec[axis] != 1)
+                        return false;
+                }
+
+                auto begin_const = ov::as_type_ptr<ov::op::v0::Constant>(begin_input);
+                if (!begin_const)
+                    return false;
+                auto end_const = ov::as_type_ptr<ov::op::v0::Constant>(end_input);
+                if (!end_const)
+                    return false;
```

Comment:
Maybe we can check this at the beginning to quickly skip nodes that don't meet our requirements?